src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,671 @@
+}
+
+export interface DropdownState {
+  value: DropdownListItem | DropdownListItem[]
+  backspaceDelete: boolean
+  focused: boolean
+  searchQuery?: string
+}
+
+export interface DropdownListItem {
+  header: string
+  key?: string
+  content?: string
+  image?: string
+}
+/**
+ * A Dropdown allows a user to select a value or a multitude of values from a number of options.
+ */
+export default class Dropdown extends AutoControlledComponent<
+  Extendable<DropdownProps>,
+  DropdownState
+> {
+  private inputRef: HTMLElement
+
+  static displayName = 'Dropdown'
+
+  static className = 'ui-dropdown'
+
+  static propTypes = {
+    ...commonUIComponentPropTypes,
+    defaultSearchQuery: PropTypes.string,
+    defaultValue: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+    fluid: PropTypes.bool,
+    getA11yStatusMessage: PropTypes.func,
+    getA11ySelectedMessage: PropTypes.func,
+    getA11yRemovedMessage: PropTypes.func,
+    getA11yRemoveItemMessage: PropTypes.func,
+    items: PropTypes.arrayOf(PropTypes.object),
+    itemToString: PropTypes.func,
+    multiple: PropTypes.bool,
+    noResultsMessage: PropTypes.string,
+    onBackspaceDelete: PropTypes.func,
+    onCloseIconClick: PropTypes.func,
+    onCloseIconKeyDown: PropTypes.func,
+    onContainerClick: PropTypes.func,
+    onDropdownChange: PropTypes.func,
+    onInputBlur: PropTypes.func,
+    onInputFocus: PropTypes.func,
+    onInputKeyDown: PropTypes.func,
+    onLabelClick: PropTypes.func,
+    onSearchQueryChange: PropTypes.func,
+    placeholder: PropTypes.string,
+    search: PropTypes.bool,
+    searchQuery: PropTypes.string,
+    toggleButton: PropTypes.bool,
+    value: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+  }
+
+  static defaultProps = {
+    as: 'div',
+  }
+
+  static autoControlledProps = ['searchQuery', 'value']
+
+  state: DropdownState = {
+    backspaceDelete: this.props.multiple ? true : undefined,
+    focused: false,
+    searchQuery: this.props.search ? '' : undefined,
+    value: this.props.multiple ? [] : null,
+  }
+
+  public renderComponent({
+    ElementType,
+    classes,
+    styles,
+    variables,
+    rest,
+  }: RenderResultConfig<DropdownProps>) {
+    const { search, multiple, toggleButton, getA11yStatusMessage, itemToString } = this.props
+    const { searchQuery } = this.state
+    const optionalDownshiftProps = {
+      // in multiple dropdown, we hold active values in the array, and default active is null.
+      ...(multiple && { selectedItem: null }),
+    }
+
+    return (
+      <ElementType {...rest} className={classes.root}>
+        <Downshift
+          onChange={this.handleChange}
+          inputValue={searchQuery}
+          stateReducer={this.stateReducer}
+          getA11yStatusMessage={getA11yStatusMessage || this.getA11yStatusMessage}
+          itemToString={
+            itemToString ? itemToString : (item: DropdownListItem) => (item ? item.header : '')
```

Comment:
can be simplified a bit
```js
itemToString || (item: DropdownListItem) => (item ? item.header : '')
```

Reply:
`itemToString={itemToString || ((item: DropdownListItem) => (item ? item.header : ''))}`